parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -781,6 +910,20 @@ pub mod pallet {
 			});
 
 			Self::deposit_event(Event::BeaconHeaderImported { block_hash: block_root, slot });
+
+			Ok(())
+		}
+
+		fn add_finalized_header_slot(slot: u64) -> DispatchResult {
+			<FinalizedBeaconHeaderSlots<T>>::try_mutate(|b_vec| {
+				if b_vec.len() as u32 == T::MaxFinalizedHeaderSlotArray::get() {
```

Comment:
perhaps `T::MaxFinalizedHeaderSlotArray` should be u64 too? If in doubt.

I think anywhere we are casting from u32 to u64, perhaps the original value should be u64 to being with.